services/worker/src/worker/job_runners/dataset/split_names_from_streaming.py

```diff
@@ -0,0 +1,168 @@
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+class ResponseNotReadyError(DatasetSplitNamesFromStreamingJobRunnerError):
+    """Raised when the response has not been processed yet from any source."""
+
+    def __init__(self, message: str):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "ResponseNotReady")
+
+
+class FailedConfigItem(ConfigItem):
+    error: Mapping[str, Any]
+
+
+class DatasetSplitNamesFromStreamingResponse(TypedDict):
+    splits: List[SplitItem]
+    pending: List[ConfigItem]
+    failed: List[FailedConfigItem]
+
+
+def compute_dataset_split_names_from_streaming_response(
+    dataset: str,
+) -> Tuple[DatasetSplitNamesFromStreamingResponse, float]:
+    """
+    Get the response of /splits for one specific dataset on huggingface.co
+    computed from responses cached in /split-names-from-streaming step.
+
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated by a `/`.
+    Returns:
+        `DatasetSplitNamesFromStreamingResponse`: An object with a list of split names for the dataset [splits],
+         a list of pending configs to be processed [pending] and the list of errors [failed] by config.
+    <Tip>
+    Raises the following errors:
+        - [`~job_runners.dataset_split_names_from_streaming.PreviousStepStatusError`]
+          If the the previous step gave an error.
+        - [`~job_runners.dataset_split_names_from_streaming.PreviousStepFormatError`]
+            If the content of the previous step has not the expected format
+        - [`~libcommon.dataset.DatasetNotFoundError`]
+            If previous step content was not found for the dataset
+    </Tip>
+    """
+    logging.info(f"get dataset split names from dataset info for dataset={dataset}")
+    try:
+        config_names = get_response(kind="/config-names", dataset=dataset)
+        config_content = config_names["content"]["config_names"]
+    except DoesNotExist as e:
+        raise DatasetNotFoundError("No response found in previous step '/config-names' for this dataset.", e) from e
+    except KeyError as e:
+        raise PreviousStepFormatError("Previous step '/config-names' did not return the expected content.") from e
+
+    if config_names["http_status"] != HTTPStatus.OK:
+        raise PreviousStepStatusError(
+            f"Previous step gave an error: {config_names['http_status']}. This job should not have been created."
+        )
+    try:
+        splits: List[SplitItem] = []
+        pending: List[ConfigItem] = []
+        failed: List[FailedConfigItem] = []
+        total = 0
+        for config_item in config_content:
+            config = config_item["config"]
+            total += 1
+            try:
+                response = get_response(kind="/split-names-from-streaming", dataset=dataset, config=config)
+            except DoesNotExist:
```

Comment:
in that case, should we look if a pending job exists (WAITING or STARTED), and create it if not?

Reply:
Yes, I think that is a good idea, we should apply it for all the aggregates job runners, is it ok if I create a new issue that will cover all the existing runners that manages the same scenario? (Parquet, size, splits)

Reply:
yes, better in a new issue.

By the way, it should not be done by the method we use to get the cache: it should return the content, and if missing, check if the cache should have existed. It's somewhat related to https://github.com/huggingface/datasets-server/issues/751 (having the cache in a microservice - it would be in charge of doing the check)

Reply:
I created the issue https://github.com/huggingface/datasets-server/issues/943